cleanlab_studio/internal/api/api.py

```diff
@@ -329,3 +333,64 @@ def poll_progress(
             res = request_function(progress_id)
         pbar.update(float(1) - pbar.n)
     return res
+
+
+def upload_predict_batch(api_key: str, model_id: str, batch: io.StringIO) -> str:
+    """Uploads prediction batch and returns query ID."""
+    url = f"{model_base_url}/{model_id}/upload"
+    res = requests.post(
+        url,
+        headers=_construct_headers(api_key),
+    )
+
+    handle_api_error(res)
+    presigned_url = res.json()["upload_url"]
+    query_id: str = res.json()["query_id"]
+    modality = res.json()["modality"]
+    header = res.json()["header"]
+    if modality == "text":
+        header_io = io.StringIO(header)
+        input_batch = io.StringIO("\n".join(chain(header_io, batch)))
```

Comment:
see comment in backend

Reply:
Got it. I'll change the lambda function to take care of this. Didn't want to touch lambda as much as possible since it's not in a state that's easy to fix and deploy right now.